core/templates/dev/head/pages/exploration_player/AudioControlsDirective.js

```diff
@@ -174,15 +162,41 @@ oppia.directive('audioControls', [
               AudioPlayerService.getCurrentAudioControlsDirectiveId();
           };
 
+          var onFinishedLoadingAudio = function(loadedFilename) {
+            if (audioRequestIsPending &&
+                loadedFilename === 
+                  AudioPreloaderService
+                    .getMostRecentlyRequestedAudioOnCurrentCard()) {
+              audioRequestIsPending = false;
+              AudioPlayerService.load(loadedFilename, directiveId)
+                .then(function() {
+                  AudioPlayerService.play();
+                });
+            }
+            $scope.audioIsLoading = false;
+          };
+
+          AudioPreloaderService.setAudioLoadedCallback(onFinishedLoadingAudio);
+
           var loadAndPlayAudioTranslation = function() {
             $scope.audioIsLoading = true;
+            audioRequestIsPending = true;
             var audioTranslation = getAudioTranslationInCurrentLanguage();
+            AudioPreloaderService.setMostRecentlyRequestedAudioOnCurrentCard(
```

Comment:
Do you need the "on current card" bit in the name? The preloader service shouldn't really have the idea of being scoped to a specific card.

Reply:
Yeah, I wasn't sure about putting that. I thought it might be slightly misleading without it just because if you switch cards the value of the field should be set to null. But I think with it it kind of just becomes too much of a mouthful to comprehend so I think I'll get rid of it.

Reply:
Removed